Mathlib/Algebra/Algebra/Subalgebra/Pointwise.lean

```diff
@@ -31,8 +31,7 @@ theorem mul_toSubmodule_le (S T : Subalgebra R A) :
 
 /-- As submodules, subalgebras are idempotent. -/
 @[simp]
-theorem mul_self (S : Subalgebra R A) : (Subalgebra.toSubmodule S) * (Subalgebra.toSubmodule S)
-    = (Subalgebra.toSubmodule S) := by
+theorem mul_self (S : Subalgebra R A) : IsIdempotentElem (Subalgebra.toSubmodule S) := by
```

Comment:
I think `mul_self` is no longer a good name

Reply:
I've renamed `mul_self` to `isIdempotentElem_subalgebra` and added a deprecated alias.